src/workerd/util/sqlite.c++

```diff
@@ -182,19 +217,182 @@ kj::Own<sqlite3_stmt> SqliteDatabase::prepareSql(sqlite3* db, kj::StringPtr sqlC
+
+    case SQLITE_CREATE_INDEX       :   /* Index Name      Table Name      */
+    case SQLITE_DROP_INDEX         :   /* Index Name      Table Name      */
+    case SQLITE_CREATE_TRIGGER     :   /* Trigger Name    Table Name      */
+    case SQLITE_DROP_TRIGGER       :   /* Trigger Name    Table Name      */
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1)) &&
+             regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
+
+    case SQLITE_TRANSACTION        :   /* Operation       NULL            */
+      {
+        // Verify param1 is one of the values we expect.
+        kj::StringPtr op = KJ_ASSERT_NONNULL(param1);
+        KJ_ASSERT(op == "BEGIN" || op == "ROLLBACK" || op == "COMMIT", op);
+      }
+      KJ_ASSERT(param2 == nullptr);
+      return true;
+
+    case SQLITE_SAVEPOINT          :   /* Operation       Savepoint Name  */
+      {
+        // Verify param1 is one of the values we expect.
+        kj::StringPtr op = KJ_ASSERT_NONNULL(param1);
+        KJ_ASSERT(op == "BEGIN" || op == "ROLLBACK" || op == "COMMIT", op);
+      }
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
+
+    case SQLITE_PRAGMA             :   /* Pragma Name     1st arg or NULL */
+      // TODO(sqlite): Decide which pragmas are OK.
+      return false;
+
+    case SQLITE_FUNCTION           :   /* NULL            Function Name   */
+      // TODO(sqlite): Decide which function are OK.
+      return false;
+
+    // ---------------------------------------------------------------
+    // Stuff that is never allowed
+
+    case SQLITE_CREATE_VTABLE      :   /* Table Name      Module Name     */
+    case SQLITE_DROP_VTABLE        :   /* Table Name      Module Name     */
+      // Virtual tables are tables backed by some native-code callbacks. We don't support these.
+      return false;
+
+    case SQLITE_ATTACH             :   /* Filename        NULL            */
+    case SQLITE_DETACH             :   /* Database Name   NULL            */
+      // We do not support attached databases. It seems unlikely that we ever will.
+      return false;
+
+    case SQLITE_CREATE_TEMP_TABLE  :   /* Table Name      NULL            */
+    case SQLITE_DROP_TEMP_TABLE    :   /* Table Name      NULL            */
+    case SQLITE_CREATE_TEMP_INDEX  :   /* Index Name      Table Name      */
+    case SQLITE_DROP_TEMP_INDEX    :   /* Index Name      Table Name      */
+    case SQLITE_CREATE_TEMP_TRIGGER:   /* Trigger Name    Table Name      */
+    case SQLITE_DROP_TEMP_TRIGGER  :   /* Trigger Name    Table Name      */
+    case SQLITE_CREATE_TEMP_VIEW   :   /* View Name       NULL            */
+    case SQLITE_DROP_TEMP_VIEW     :   /* View Name       NULL            */
+      // TODO(someday): Allow temporary tables. Creating a temporary table actually causes
+      //   SQLite to open a separate temporary file to place the data in. Currently, our storage
+      //   engine has no support for this.
+      return false;
```

Comment:
Can a temporary table outlive a transaction? I'm wondering if we need the storage system to do anything but provide access to a new, ephemeral file on disk?